# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/COLRFonts.h
# Commit: 673f9010ea75
# Full Hash: 673f9010ea75382463067acf3da66683f4eecae1
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-08-17 09:10:29
# Regressor Bug: 1740530
# File Overlap Count: 1
# Description:
#   Bug 1740530 - patch 3 - Replace GetColorGlyphLayers and the layer-rendering code in gfxFont with a zero-copy implementation in COLRFonts. r=gfx-reviewers,lsalzman
#   
#   Depends on D152037
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152038
# ==============================================================================

diff -r 429601610f26 -r 673f9010ea75 gfx/thebes/COLRFonts.h
--- a/gfx/thebes/COLRFonts.h	Tue Aug 16 12:48:32 2022 +0000
+++ b/gfx/thebes/COLRFonts.h	Tue Aug 16 12:48:32 2022 +0000
@@ -6,27 +6,36 @@
 #ifndef COLR_FONTS_H
 #define COLR_FONTS_H
 
-#include "nsTArray.h"
+#include "mozilla/gfx/2D.h"
 
 struct hb_blob_t;
 
 namespace mozilla {
+
+namespace layout {
+class TextDrawTarget;
+}
+
 namespace gfx {
-struct DeviceColor;
+struct COLRBaseGlyphRecord;
 
 class COLRFonts {
  public:
   // for color layer from glyph using COLR and CPAL tables
   static bool ValidateColorGlyphs(hb_blob_t* aCOLR, hb_blob_t* aCPAL);
-  static bool GetColorGlyphLayers(
-      hb_blob_t* aCOLR, hb_blob_t* aCPAL, uint32_t aGlyphId,
-      const mozilla::gfx::DeviceColor& aDefaultColor,
-      nsTArray<uint16_t>& aGlyphs,
-      nsTArray<mozilla::gfx::DeviceColor>& aColors);
-  static bool HasColorLayersForGlyph(hb_blob_t* aCOLR, uint32_t aGlyphId);
+
+  static const COLRBaseGlyphRecord* GetGlyphLayers(hb_blob_t* aCOLR,
+                                                   uint32_t aGlyphId);
+
+  static bool PaintGlyphLayers(
+      hb_blob_t* aCOLR, hb_blob_t* aCPAL, const COLRBaseGlyphRecord* aLayers,
+      DrawTarget* aDrawTarget, layout::TextDrawTarget* aTextDrawer,
+      ScaledFont* aScaledFont, DrawOptions aDrawOptions,
+      const sRGBColor& aCurrentColor, const Point& aPoint);
 };
 
 }  // namespace gfx
+
 }  // namespace mozilla
 
 #endif  // COLR_FONTS_H